File: assets/zsjz/biz/js/tag/WmDropmenu.js
require(["domReady!","wmWebUI","jcl","mobile"],function(doc,WmWebUI,$,Mobile){
	$("#show").on("tap",function(){
		var dropmenu=WmWebUI.select("dropmenu01");
		var that = $(this);
		if(dropmenu.invisible()){
			dropmenu.show();
			that.html("关闭菜单");
		}else{
			dropmenu.hidden();
			that.html("显示菜单");
		}
	});
	$("#setLabel").on("tap",function(){
		var dropmenu=WmWebUI.select("dropmenu01");
		var str=prompt("请输入菜单名。示例：我的活动");
		if(str == ""){
			str = "默认名称";
		}
		dropmenu.setLabel(str);
	});
	$("#setAction").on("tap", function(){
		var dropmenu=WmWebUI.select("dropmenu01");
		var myDate=new Date();
		dropmenu.setOpenAction(function(){
			Mobile.tip("正在打开。事件设置时间："+myDate.toLocaleString());
		});
		dropmenu.setCloseAction(function(){
			Mobile.tip("正在关闭。事件设置时间："+myDate.toLocaleString());
		});
		Mobile.tip("事件设置成功。设置时间："+myDate.toLocaleString());
	});
	$("#cleanAction").on("tap",function(){
		var dropmenu=WmWebUI.select("dropmenu01");
		dropmenu.setOpenAction(function(){
		});
		dropmenu.setCloseAction(function(){
		});
		alert("事件清除成功！");
	});
	$("#setItemAction").on("tap",function(){
		var dropmenu=WmWebUI.select("dropmenu01");
		var items=dropmenu.getItems();
		var myDate=new Date();
		if(items.length>0){
			items[0].click(function(){
				alert("我是第一个菜单项，事件设置时间为："+myDate.toLocaleString());
			});
			Mobile.tip("事件设置成功。设置时间："+myDate.toLocaleString());
		}else{
			alert("事件设置失败，没有菜单项。");
		}
	});
	$("#removeAll").on("tap",function(){
		var dropmenu=WmWebUI.select("dropmenu01");
		dropmenu.removeAll();
		alert("清除成功。");
	});
/*	$("#push").on("tap",function(){
		var dropmenu=WmWebUI.select("dropmenu01");
		var str=prompt("请输入菜单显示条目。示例：中国");
		dropmenu.push(str);
		alert("添加成功。")
	});*/
	$("#remove").on("tap",function(){
		var dropmenu=WmWebUI.select("dropmenu01");
		var items=dropmenu.getItems();
		items[0].remove();
		alert("删除成功");
	});
	$("#html").on("tap",function(){
		var dropmenu=WmWebUI.select("dropmenu01");
		var items=dropmenu.getItems();
		var str=prompt("第一个菜单项的内容。示例：<b>中国</b>")
		items[0].html(str);
		alert("添加成功");
	});
	new IScroll("#content",{tap:true,scrollbars: true});
});

File: assets/zsjz/template/lua/tag/WmDropmenu.lua
local Class = require("util.Class")
local Tag = require("engine.Tag")
local WmDropmenu = Class(Tag)

--htmlbuff
function WmDropmenu:createNew(obj,htmlbuff)
  self.htmlbuff=htmlbuff
  self.clicks = {}
end

--attr所有属性
function WmDropmenu:doStartTag(attr)
  --属性初始化
  attr.closeAction=attr.closeAction or "console.log('close')"
  attr.openAction=attr.oppenAction or "console.log('open')"
  self.attr=attr
  self.htmlbuff:append([[
<div id="]])
  self.htmlbuff:append(attr.id)
  self.htmlbuff:append([[">
      <div class="fn">
        <span class="e_button"><span>]])
  self.htmlbuff:append(attr.label)
  self.htmlbuff:append([[</span><span class="e_ico-unfold"></span></span>
      </div>
      <div class="c_dropmenu">
        <ul>]])
end

function WmDropmenu:doEndTag()
  self.htmlbuff:append([[
     </ul>
      </div>
    </div>
  ]])
    
    
 self.htmlbuff:append([[
  <script>
  require(["wmDropmenu","util"],function(WmDropmenu) {
      var wmDropmenu=new WmDropmenu(]])
 self.htmlbuff:append(self.attr.id)
 self.htmlbuff:append([[);
      wmDropmenu.setOpenAction(function(){
        ]])
 self.htmlbuff:append(self.attr.openAction)
 self.htmlbuff:append([[
      
      });
      wmDropmenu.setCloseAction(function(){
        ]])
 self.htmlbuff:append(self.attr.closeAction)
 self.htmlbuff:append([[
      
      });
      wmDropmenu.create();
      var items=wmDropmenu.getItems();
 ]])
 for k,v in pairs(self.clicks) do
    --print(k,v,type(v))
    if v~=-1 then
      self.htmlbuff:append([[
      items[]])
      self.htmlbuff:append(k-1)
      self.htmlbuff:append([[].click(function(){
          ]])
      self.htmlbuff:append(v)
      self.htmlbuff:append([[
      
        });
      ]])
    end
  end
  self.htmlbuff:append([[
  WmWebUI.store("]])
  self.htmlbuff:append(self.attr.id)
  self.htmlbuff:append([[",wmDropmenu);
});
  </script>
  ]])
end

--[[
require(["wmDropmenu","util"],function(WmDropmenu) {
  var wmDropmenu=new WmDropmenu(dropmenu01);
  wmDropmenu.create();
  wmDropmenu.setLabel("下拉菜单01");
  wmDropmenu.setCloseAction(function(){
    console.log("closeAction");
  });
  wmDropmenu.setOpenAction(function(){
    console.log("openAction");
    console.log(wmDropmenu.getLabel());
  });
  var items=wmDropmenu.getItems();
  for(var i=0;i<items.length;i++){
    console.log("index:"+items[i].getIndex());
    console.log("parent:"+items[i].getParent());
    items[i].click(function(obj){
      alert(obj.html());
    });
  }
  console.log("length:"+items.length);
  items[2].remove();
  console.log("length:"+items.length);
  for(var i=0;i<items.length;i++){
    //alert(items[i].html());
  }
  //wmDropmenu.removeAll();
  var item1=wmDropmenu.push("111");
  item1.click(function(obj){
    alert("我是新来的1");
  });
  var item2=wmDropmenu.push("22222");
  item2.click(function(obj){
    alert("我是新来的2");
  });
  item1.remove();
  //wmDropmenu.getItems()[0].remove();
  $("#testbtn").click(function(){
    if(wmDropmenu.Invisible()){
      wmDropmenu.show();
    }else{
      wmDropmenu.hidden();
    }
  });
});

]]
function WmDropmenu:addClicks(click)
  table.insert(self.clicks, click)
end


return WmDropmenu
